__tests__/friend.test.js

```diff
@@ -0,0 +1,76 @@
+/**
+* @jest-environment jsdom
+*/
+
+import React from 'react';
+import {rest} from 'msw';
+import {setupServer} from 'msw/node';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Friend from '../client/src/components/Friend.jsx';
+
+// test api results
+const apiMockResults = [
+  { 'id': '1', 'username': 'graphite-x' },
+  { 'id': '2', 'username': 'gryffindor' },
+  { 'id': '3', 'username': 'tygah' }
+];
+
+const server = setupServer(
+  rest.get('/api/getUsers', (req, res, ctx) => {
+    const searchParam = req.url.searchParams.get('username');
+    const responseData = apiMockResults;
+
+    if (searchParam === 'gr') {
+      responseData = apiMockResults.slice(0, 2);
```

Comment:
Is this `responseData` const the variable that controls the logic for the test you were having trouble with? If so, perhaps using `let` would allow the re-assignment that you are doing since `const` can neither be re-assigned nor re-declared.